account_mass_reconcile_transaction_ref/tests/test_account_reconcile_transaction_ref.py

```diff
@@ -0,0 +1,118 @@
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo.tests.common import SavepointCase
+
+
+class TestAccountReconcileTransactionRef(SavepointCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))
+
+        cls.partner = cls.env.ref('base.res_partner_18')
+        cls.account_receivable = cls.env['account.account'].search([
+            ('user_type_id', '=',
+             cls.env.ref('account.data_account_type_receivable').id)
+        ], limit=1)
+        account_revenue = cls.env['account.account'].search([
+            ('user_type_id', '=',
+             cls.env.ref('account.data_account_type_revenue').id)
+        ], limit=1)
+        sales_journal = cls.env['account.journal'].search([
+            ('type', '=', 'sale')], limit=1)
+        # Create invoice
+        cls.cust_invoice = cls.env['account.invoice'].create({
+            'partner_id': cls.partner.id,
+            'type': 'out_invoice',
+            'account_id': cls.account_receivable.id,
+            'journal_id': sales_journal.id,
+            'invoice_line_ids': [(0, 0, {
+                'name': '[CONS_DEL01] Server',
+                'product_id': cls.env.ref('product.consu_delivery_01').id,
+                'account_id': account_revenue.id,
+                'price_unit': 1000.0,
+                'quantity': 1.0,
+            })],
+            'transaction_id': 'test_transaction_id',
+        })
+        cls.cust_invoice.action_invoice_open()
+
+    def test_mass_reconcile_transaction_ref_vs_ref(self):
+        self.assertEqual(self.cust_invoice.state, 'open')
+        self.assertEqual(self.cust_invoice.transaction_id,
+                         'test_transaction_id')
+        bank_journal = self.env['account.journal'].search([
+            ('type', '=', 'sale')], limit=1)
+
+        # Create payment
+        payment = self.env['account.payment'].create({
```

Comment:
I think the payment creation can be factorized https://github.com/OCA/account-reconcile/pull/276/files#diff-2031f1d37eb509547481674f127dbd34R85